src/rkopenmdao/runge_kutta_integrator.py

```diff
@@ -625,28 +625,53 @@ def _setup_arrays(self):
 
     def _setup_runge_kutta_scheme(self):
         butcher_tableau: ButcherTableau = self.options["butcher_tableau"]
-        self._runge_kutta_scheme = RungeKuttaScheme(
-            butcher_tableau,
-            TimeStageProblemComputeFunctor(
-                self.options["time_stage_problem"],
-                self.options["integration_control"],
-                self._quantity_metadata,
-            ),
-            TimeStageProblemComputeJacvecFunctor(
-                self.options["time_stage_problem"],
-                self.options["integration_control"],
-                self._quantity_metadata,
-                self._of_vars,
-                self._wrt_vars,
-            ),
-            TimeStageProblemComputeTransposeJacvecFunctor(
-                self.options["time_stage_problem"],
-                self.options["integration_control"],
-                self._quantity_metadata,
-                self._of_vars,
-                self._wrt_vars,
-            ),
-        )
+        if not butcher_tableau.is_embedded:
```

Comment:
After some further thought, I'm not sure anymore if I like this structure. There are legitimate scenarios where an embedded scheme is used while not using adaptive time stepping, e.g. when the the non-embedded scheme is a good RK-scheme as is.  In that case, there should be the possibility to use the scheme in a time-nonadaptive way.

My proposal: Just add a flag to the RungeKuttaIntegrator that indicates that adaptive time stepping is desired. This is passed to the `RungeKuttaScheme` which then decides how to procede. The `RungeKuttaScheme` and `EmbeddedRungeKutta` should be merged. A proposal for that is in a comment at the respective file.